src/frontends/onnx/frontend/src/core/graph.cpp

```diff
@@ -58,16 +58,20 @@ bool common_node_for_all_outputs(const OutputVector& outputs) {
 }  // namespace detail
 
 Graph::Graph(const std::shared_ptr<ONNX_NAMESPACE::ModelProto>& model_proto,
-             const std::shared_ptr<ov::frontend::TelemetryExtension>& telemetry)
-    : Graph(model_proto, common::make_unique<GraphCache>(), telemetry) {}
+             const std::shared_ptr<ov::frontend::TelemetryExtension>& telemetry,
+             const std::shared_ptr<ov::frontend::ProgressReporterExtension>& progress_reporter)
+    : Graph(model_proto, common::make_unique<GraphCache>(), telemetry, progress_reporter) {}
 
 Graph::Graph(const std::shared_ptr<ONNX_NAMESPACE::ModelProto>& model_proto,
              std::unique_ptr<GraphCache>&& cache,
-             const std::shared_ptr<ov::frontend::TelemetryExtension>& telemetry)
+             const std::shared_ptr<ov::frontend::TelemetryExtension>& telemetry,
+             const std::shared_ptr<ov::frontend::ProgressReporterExtension>& progress_reporter)
     : m_model{common::make_unique<Model>(model_proto)},
       m_cache{std::move(cache)},
-      m_telemetry(telemetry) {
+      m_telemetry(telemetry),
```

Comment:
Perhaps we need some kind of container for extensions?
```
class ExtensionHolder {
     ExtensionHolder(shared_ptr<TelemetryExtension> telemetry, 
                                 shared_ptr<ProgressReporter> progress_reporter);
       get_telemetry() { //returns telemetry or nullptr }
       get_progress_reporter() { // returns reported or nullptr }
}  
```
@mbencer @tomdol What do you think?

Reply:
You've read my mind ;) That's something I'd like to add and clean this code up a bit.